### PR TITLE
ROX-11524: (Part 4/4) Migrate PermissionSet to remove AuthPlugins

### PR DESCRIPTION
**Pull request by 0x656b694d**
## Description

* Migrate permission sets
* Remove the AuthPlugins bucket

Extracted from #2507.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

Automated tests.

### CODE REVIEW

File: migrator/migrations/m_107_to_m_108_remove_auth_plugin/migration.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+			bytes, err := proto.Marshal(ps)
+			if err != nil {
+				log.WriteToStderrf("failed to marshal migrated permissionset for key %s: %v", id, err)
+				continue
+			}
+			if err := bucket.Put([]byte(id), bytes); err != nil {
+				return err
+			}
+		}
+		return tx.DeleteBucket(authPluginBucket)
```
Comment (rhybrillou) [inline migrator/migrations/m_107_to_m_108_remove_auth_plugin/migration.go:78]: I'd rather isolate the two actions performed by the migration in dedicated functions: one for the auth plugin cleanup and another for the removal of the AuthPlugin reference from the permission sets.

**Review by rhybrillou — DISMISSED** [review]
Please isolate the migration part for `authzPlugins` (Bolt) and `permission_sets` (RocksDB)

File: migrator/migrations/m_107_to_m_108_remove_auth_plugin/migration.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+			return nil
+		},
+	}
+
+	psBucket               = []byte("permission_sets")
+	authPluginBucket       = []byte("authzPlugins")
+	AuthPluginResourceName = "AuthPlugin"
+)
+
+func migratePS(db *bbolt.DB) error {
```
[the remark this answers is not in the document]
Comment (0x656b694d) [reply migrator/migrations/m_107_to_m_108_remove_auth_plugin/migration.go:31]: :scream:

**Comment by rhybrillou** [comment]
Looking into the test failure:
pkg/grpc/authz/user/user.go:11:2: no required module provides package github.com/stackrox/rox/pkg/grpc/authz/internal/permissioncheck; to add it:
	go get github.com/stackrox/rox/pkg/grpc/authz/internal/permissioncheck

Can you make sure the import is removed if it is not needed anymore ? I think the consumer code was removed in the previous PR.

**Comment by janisz** [comment]
@rhybrillou Fixed! Let's wait for CI

**Review by janisz — APPROVED** [review]

**Pull request merged**